diff -Naur a/config-win32.h.in b/config-win32.h.in
--- a/config-win32.h.in	2008-10-06 07:22:21.000000000 +0000
+++ b/config-win32.h.in	2009-02-22 04:17:44.807557536 +0000
@@ -35,7 +35,6 @@
 
 #include <windows.h>
 #include <winsock2.h>
-#include "autodefs/defs.h"
 
 #define sleep(x) Sleep((x)*1000)
 
@@ -219,7 +218,7 @@
 #define PACKAGE @PACKAGE@
 
 /* Define to the address where bug reports for this package should be sent. */
-//#define PACKAGE_BUGREPORT "openvpn-users@lists.sourceforge.net"
+#define PACKAGE_BUGREPORT "https://bugs.torproject.org/flyspray/"
 
 /* Define to the full name of this package. */
 #define PACKAGE_NAME @PACKAGE_NAME@
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2008-10-14 04:14:36.000000000 +0000
+++ b/configure.ac	2009-02-22 04:17:44.809557232 +0000
@@ -866,6 +866,8 @@
 	openvpn.spec
 	config-win32.h
 	images/Makefile
+	tap-win32/SOURCES
+	tap-win32/i386/OemWin2k.inf
 	service-win32/Makefile
 	install-win32/Makefile
 	install-win32/settings
diff -Naur a/install-win32/maketap b/install-win32/maketap
--- a/install-win32/maketap	2008-10-07 21:10:10.000000000 +0000
+++ b/install-win32/maketap	2009-02-22 04:20:37.273338752 +0000
@@ -4,15 +4,15 @@
 # Requires the Windows DDK
 
 # get version.nsi definitions
-. autodefs/defs.sh
+#. autodefs/defs.sh
 
-if [ -n "$DDKVER" ] && [ -d "/c/WINDDK/$DDKVER" ] ; then
+if [ -d "$DDKDIR" ] ; then
 
 # common declarations for all DDK build targets
 . install-win32/ddk-common
 
 # configure tap driver sources
-MACRO="perl install-win32/macro.pl autodefs/defs.in"
+MACRO="perl install-win32/macro.pl install-win32/settings"
 IFDEF="perl install-win32/ifdef.pl"
 rm -rf tap-win32/amd64
 mkdir tap-win32/amd64
@@ -20,6 +20,9 @@
 $MACRO <tap-win32/i386/OemWin2k.inf.in | $IFDEF >tap-win32/i386/OemWin2k.inf
 $MACRO <tap-win32/i386/OemWin2k.inf.in | $IFDEF -DAMD64 >tap-win32/amd64/OemWin2k.inf
 
+# we don't use this script for building
+exit 0
+
 if [ -n "$PRODUCT_TAP_DEBUG" ] ; then
     w2ktarget="w2k c"
     amdtarget="chk $x64_tag WNET"
@@ -96,5 +99,5 @@
 fi
 
 else
-    echo Not building TAP driver -- DDK version $DDKVER NOT FOUND
+    echo Not building TAP driver -- DDK installation at \"$DDKDIR\" NOT FOUND
 fi
diff -Naur a/tap-win32/SOURCES.in b/tap-win32/SOURCES.in
--- a/tap-win32/SOURCES.in	2008-09-30 02:34:09.000000000 +0000
+++ b/tap-win32/SOURCES.in	2009-02-22 04:17:44.810557080 +0000
@@ -4,7 +4,7 @@
 MAJORCOMP=ntos
 MINORCOMP=ndis
 
-TARGETNAME=@@PRODUCT_TAP_ID@@
+TARGETNAME=tortap91
 TARGETTYPE=DRIVER
 TARGETPATH=.
 TARGETLIBS=$(DDK_LIB_PATH)\ndis.lib $(DDK_LIB_PATH)\ntstrsafe.lib
@@ -14,8 +14,9 @@
 # TAP_WIN32_MIN_x values defined in
 # config-win32.h
 C_DEFINES=
-C_DEFINES=$(C_DEFINES) -DTAP_DRIVER_MAJOR_VERSION=@@PRODUCT_TAP_MAJOR_VER@@
-C_DEFINES=$(C_DEFINES) -DTAP_DRIVER_MINOR_VERSION=@@PRODUCT_TAP_MINOR_VER@@
+C_DEFINES=$(C_DEFINES) -DTAP_DRIVER_MAJOR_VERSION=9
+C_DEFINES=$(C_DEFINES) -DTAP_DRIVER_MINOR_VERSION=1
+C_DEFINES=$(C_DEFINES) -DDDKVER_MAJOR=3790
 
 # Produce the same symbolic information for both free & checked builds.
 # This will allow us to perform full source-level debugging on both
diff -Naur a/tap-win32/i386/OemWin2k.inf.in b/tap-win32/i386/OemWin2k.inf.in
--- a/tap-win32/i386/OemWin2k.inf.in	2008-10-06 07:22:21.000000000 +0000
+++ b/tap-win32/i386/OemWin2k.inf.in	2009-02-22 04:17:44.811556928 +0000
@@ -6,9 +6,9 @@
 ; ****************************************************************************
 
 ; SYNTAX CHECKER
-; cd \WINDDK\3790\tools\chkinf
+; cd \WINDDK\{VER}\tools\chkinf
 ; chkinf c:\src\openvpn\tap-win32\i386\oemwin2k.inf
-; OUTPUT -> file:///c:/WINDDK/3790/tools/chkinf/htm/c%23+src+openvpn+tap-win32+i386+__OemWin2k.htm
+; OUTPUT -> file:///c:/WINDDK/{VER}/tools/chkinf/htm/c%23+src+openvpn+tap-win32+i386+__OemWin2k.htm
 
 ; INSTALL/REMOVE DRIVER
 ;   tapinstall install OemWin2k.inf TAP0901
@@ -48,35 +48,34 @@
 
 [Version]
    Signature = "$Windows NT$"
-   CatalogFile = @@PRODUCT_TAP_ID@@.cat
    ClassGUID = {4d36e972-e325-11ce-bfc1-08002be10318}
    Provider = %Provider%
    Class = Net
 
 ; This version number should match the version
 ; number given in SOURCES.
-   DriverVer=@@PRODUCT_TAP_RELDATE@@,@@PRODUCT_TAP_MAJOR_VER@@.00.00.@@PRODUCT_TAP_MINOR_VER@@
+   DriverVer=09/26/2009,9.00.00.1
 
 [Strings]
-   DeviceDescription = "@@PRODUCT_TAP_DEVICE_DESCRIPTION@@"
-   Provider = "@@PRODUCT_TAP_PROVIDER@@"
+   DeviceDescription = "Tor VM TAP-Win32 Adapter V9"
+   Provider = "Tor TAP-Win32 Provider"
 
 ;----------------------------------------------------------------
 ;                      Manufacturer + Product Section (Done)
 ;----------------------------------------------------------------
 [Manufacturer]
 !ifdef AMD64
-   %Provider% = @@PRODUCT_TAP_ID@@, NTamd64
+   %Provider% = tortap91, NTamd64
 !else
-   %Provider% = @@PRODUCT_TAP_ID@@
+   %Provider% = tortap91
 !endif
 
 !ifdef AMD64
-[@@PRODUCT_TAP_ID@@.NTamd64]
+[tortap91.NTamd64]
 !else
-[@@PRODUCT_TAP_ID@@]
+[tortap91]
 !endif
-   %DeviceDescription% = @@PRODUCT_TAP_ID@@.ndi, @@PRODUCT_TAP_ID@@
+   %DeviceDescription% = tortap91.ndi, tortap91
 
 ;---------------------------------------------------------------
 ;                             Driver Section (Done)
@@ -91,23 +90,23 @@
 ;    NCF_HAS_UI = 0x80
 ;----------------- Characteristics ------------
 
-[@@PRODUCT_TAP_ID@@.ndi]
-   CopyFiles       = @@PRODUCT_TAP_ID@@.driver, @@PRODUCT_TAP_ID@@.files
-   AddReg          = @@PRODUCT_TAP_ID@@.reg
-   AddReg          = @@PRODUCT_TAP_ID@@.params.reg
-   Characteristics = @@PRODUCT_TAP_CHARACTERISTICS@@
+[tortap91.ndi]
+   CopyFiles       = tortap91.driver
+   AddReg          = tortap91.reg
+   AddReg          = tortap91.params.reg
+   Characteristics = 0x01
 
-[@@PRODUCT_TAP_ID@@.ndi.Services]
-   AddService = @@PRODUCT_TAP_ID@@,        2, @@PRODUCT_TAP_ID@@.service
+[tortap91.ndi.Services]
+   AddService = tortap91,        2, tortap91.service
 
-[@@PRODUCT_TAP_ID@@.reg]
-   HKR, Ndi,            Service,      0, "@@PRODUCT_TAP_ID@@"
+[tortap91.reg]
+   HKR, Ndi,            Service,      0, "tortap91"
    HKR, Ndi\Interfaces, UpperRange,   0, "ndis5"
    HKR, Ndi\Interfaces, LowerRange,   0, "ethernet"
    HKR, ,               Manufacturer, 0, "%Provider%"
    HKR, ,               ProductName,  0, "%DeviceDescription%"
 
-[@@PRODUCT_TAP_ID@@.params.reg]
+[tortap91.params.reg]
    HKR, Ndi\params\MTU,                  ParamDesc, 0, "MTU"
    HKR, Ndi\params\MTU,                  Type,      0, "int"
    HKR, Ndi\params\MTU,                  Default,   0, "1500"
@@ -148,13 +147,13 @@
 ;    SERVICE_DISABLED     = 0x4
 ;---------- Start Mode ---------------
 
-[@@PRODUCT_TAP_ID@@.service]
+[tortap91.service]
    DisplayName = %DeviceDescription%
    ServiceType = 1
    StartType = 3
    ErrorControl = 1
    LoadOrderGroup = NDIS
-   ServiceBinary = %12%\@@PRODUCT_TAP_ID@@.sys
+   ServiceBinary = %12%\tortap91.sys
 
 ;-----------------------------------------------------------------
 ;                                File Installation
@@ -165,30 +164,17 @@
 ;    COPYFLG_NOVERSIONCHECK = 0x04
 ;----------------- Copy Flags ------------
 
-; SourceDisksNames
-; diskid = description[, [tagfile] [, <unused>, subdir]]
-; 1 = "Intel Driver Disk 1",e100bex.sys,,
-
 [SourceDisksNames]
-   1 = %DeviceDescription%, @@PRODUCT_TAP_ID@@.sys
-
-; SourceDisksFiles
-; filename_on_source = diskID[, [subdir][, size]]
-; e100bex.sys = 1,, ; on distribution disk 1
+   1 = %DeviceDescription%, tortap91.sys
 
 [SourceDisksFiles]
-@@PRODUCT_TAP_ID@@.sys = 1
+   tortap91.sys = 1
 
 [DestinationDirs]
-   @@PRODUCT_TAP_ID@@.files  = 11
-   @@PRODUCT_TAP_ID@@.driver = 12
-
-[@@PRODUCT_TAP_ID@@.files]
-;   TapPanel.cpl,,,6   ; COPYFLG_NOSKIP | COPYFLG_NOVERSIONCHECK
-;   cipsrvr.exe,,,6     ; COPYFLG_NOSKIP | COPYFLG_NOVERSIONCHECK
+   tortap91.driver = 12
 
-[@@PRODUCT_TAP_ID@@.driver]
-   @@PRODUCT_TAP_ID@@.sys,,,6     ; COPYFLG_NOSKIP | COPYFLG_NOVERSIONCHECK
+[tortap91.driver]
+   tortap91.sys,,,6     ; COPYFLG_NOSKIP | COPYFLG_NOVERSIONCHECK
 
 ;---------------------------------------------------------------
 ;                                      End
diff -Naur a/tap-win32/resource.rc b/tap-win32/resource.rc
--- a/tap-win32/resource.rc	2008-10-06 07:28:13.000000000 +0000
+++ b/tap-win32/resource.rc	2009-02-22 04:22:10.483168704 +0000
@@ -33,8 +33,8 @@
 				VFT2_DRV_COMM
 */
 
-#define VER_COMPANYNAME_STR         "The OpenVPN Project"
-#define VER_FILEDESCRIPTION_STR     "TAP-Win32 Virtual Network Driver"
+#define VER_COMPANYNAME_STR         "The Tor Project, Inc."
+#define VER_FILEDESCRIPTION_STR     "Tor VM TAP-Win32 Virtual Network Driver"
 #define VER_ORIGINALFILENAME_STR    TAP_COMPONENT_ID ".sys"
 #define VER_LEGALCOPYRIGHT_YEARS    "2003-2008"
 #define VER_LEGALCOPYRIGHT_STR      "OpenVPN Technologies, Inc."
diff -Naur a/tap-win32/tapdrvr.c b/tap-win32/tapdrvr.c
--- a/tap-win32/tapdrvr.c	2008-10-06 07:22:21.000000000 +0000
+++ b/tap-win32/tapdrvr.c	2009-02-22 04:17:44.919540512 +0000
@@ -39,10 +39,10 @@
 // TAP_IOCTL_CONFIG_TUN ioctl.
 //======================================================
 
-#include "../../autodefs/defs.h"
 #ifndef DDKVER_MAJOR
 #error DDKVER_MAJOR must be defined as the major number of the DDK Version
 #endif
+#define PRODUCT_TAP_DEVICE_DESCRIPTION "Tor VM TAP-Win32 Device Driver"
 
 #define NDIS_MINIPORT_DRIVER
 #define BINARY_COMPATIBLE 0
diff -Naur a/version.m4 b/version.m4
--- a/version.m4	2008-11-19 18:08:50.000000000 +0000
+++ b/version.m4	2009-02-22 04:24:31.835679856 +0000
@@ -1,6 +1,11 @@
-dnl define the OpenVPN version
+define(PRODUCT_NAME,[TorTap32])
 define(PRODUCT_VERSION,[2.1_rc15])
-dnl define the TAP version
-define(PRODUCT_TAP_ID,[tap0901])
+define(PRODUCT_TAP_ID,[tortap91])
 define(PRODUCT_TAP_WIN32_MIN_MAJOR,[9])
 define(PRODUCT_TAP_WIN32_MIN_MINOR,[1])
+define(PRODUCT_TAP_MAJOR_VER,[9])
+define(PRODUCT_TAP_MINOR_VER,[4])
+define(PRODUCT_TAP_RELDATE,[02/22/2009])
+define(PRODUCT_TAP_DEVICE_DESCRIPTION,[Tor VM TAP-Win32 Adapter V9])
+define(PRODUCT_TAP_PROVIDER,[Tor VM TAP-Win32 Provider V9])
+define(PRODUCT_TAP_CHARACTERISTICS,[0x81])
