diff -Naur a/src/network/ssl/qsslsocket_openssl_symbols.cpp b/src/network/ssl/qsslsocket_openssl_symbols.cpp
--- a/src/network/ssl/qsslsocket_openssl_symbols.cpp	2009-02-25 21:32:38.000000000 +0000
+++ b/src/network/ssl/qsslsocket_openssl_symbols.cpp	2009-03-06 16:14:07.852583592 +0000
@@ -320,16 +320,24 @@
 # ifdef Q_OS_WIN
     QLibrary *ssleay32 = new QLibrary(QLatin1String("ssleay32"));
     if (!ssleay32->load()) {
-        // Cannot find ssleay32.dll
         delete ssleay32;
-        return pair;
+        ssleay32 = new QLibrary(QLatin1String("ssleay32-0.9.8"));
+        if (!ssleay32->load()) {
+            // Cannot find ssleay32.dll
+            delete ssleay32;
+            return pair;
+        }
     }
 
     QLibrary *libeay32 = new QLibrary(QLatin1String("libeay32"));
     if (!libeay32->load()) {
-        delete ssleay32;
         delete libeay32;
-        return pair;
+        libeay32 = new QLibrary(QLatin1String("cryptoeay32-0.9.8"));
+        if (!libeay32->load()) {
+            delete ssleay32;
+            delete libeay32;
+            return pair;
+        }
     }
 
     pair.first = ssleay32;
